src/basic/in-addr-util.h

```diff
@@ -68,14 +69,45 @@ int in_addr_prefix_range(
                 unsigned prefixlen,
                 union in_addr_union *ret_start,
                 union in_addr_union *ret_end);
+
 int in_addr_to_string(int family, const union in_addr_union *u, char **ret);
 static inline int in6_addr_to_string(const struct in6_addr *u, char **ret) {
         return in_addr_to_string(AF_INET6, (const union in_addr_union*) u, ret);
 }
-int in_addr_prefix_to_string(int family, const union in_addr_union *u, unsigned prefixlen, char **ret);
-static inline int in6_addr_prefix_to_string(const struct in6_addr *u, unsigned prefixlen, char **ret) {
-        return in_addr_prefix_to_string(AF_INET6, (const union in_addr_union*) u, prefixlen, ret);
+
+/* Note: the lifetime of the compound literal is the immediately surrounding block,
+ * see C11 §6.5.2.5, and
+ * https://stackoverflow.com/questions/34880638/compound-literal-lifetime-and-if-blocks */
+#define IN_ADDR_MAX CONST_MAX(INET_ADDRSTRLEN, INET6_ADDRSTRLEN)
+#define IN_ADDR_TO_STRING(family, addr) ASSERT_PTR(inet_ntop(family, addr, (char[IN_ADDR_MAX]){}, IN_ADDR_MAX))
+#define IN4_ADDR_TO_STRING(addr) IN_ADDR_TO_STRING(AF_INET, addr)
+#define IN6_ADDR_TO_STRING(addr) IN_ADDR_TO_STRING(AF_INET6, addr)
+
+int in_addr_prefix_to_string(
+                int family,
+                const union in_addr_union *u,
+                unsigned prefixlen,
+                char *buf,
+                size_t buf_len);
+
+static inline const char* _in_addr_prefix_to_string(
+                int family,
+                const union in_addr_union *u,
+                unsigned prefixlen,
+                char *buf,
+                size_t buf_len) {
+        int r = in_addr_prefix_to_string(family, u, prefixlen, buf, buf_len);
```

Comment:
Maybe drop `r` altogether and do just

```c
       if (in_addr_prefix_to_string(family, u, prefixlen, buf, buf_len) < 0)
              return NULL;
```
?

Reply:
Ah, good point. I'll adjust it like that on the next push.